Sources/Signal.swift

```diff
@@ -2015,177 +1876,346 @@ private enum ThrottleWhileState<Value> {
 	}
 }
 
+private protocol SignalAggregateStrategy {
+	/// Update the latest value of the signal at `position` to be `value`.
+	///
+	/// - parameters:
+	///   - value: The latest value emitted by the signal at `position`.
+	///   - position: The position of the signal.
+	///
+	/// - returns: `true` if the aggregating signal should terminate as a result of the
+	///            update. `false` otherwise.
+	mutating func update(_ value: Any, at position: Int) -> Bool
+
+	/// Record the completion of the signal at `position`.
+	///
+	/// - parameters:
+	///   - position: The position of the signal.
+	///
+	/// - returns: `true` if the aggregating signal should terminate as a result of the
+	///            completion. `false` otherwise.
+	mutating func complete(at position: Int) -> Bool
+
+	init(count: Int, action: @escaping (ContiguousArray<Any>) -> Void)
+}
+
 extension Signal {
+	private struct CombineLatestStrategy: SignalAggregateStrategy {
+		private enum Placeholder {
+			case none
+		}
+
+		private var values: ContiguousArray<Any>
+		private var completionCount: Int
+		private let action: (ContiguousArray<Any>) -> Void
+
+		private var _hasAllSentInitial: Bool
+		private var hasAllSentInitial: Bool {
```

Comment:
This reads a little funny. Maybe rename to `haveAllSentInitial` or `hasAllInitialValues`?